### DISCUSSION TITLE
Custom ordering of training examples within an epoch

### DISCUSSION THREAD

**Discussion by awasthiabhijeet** [allenai/allennlp#5382] [Q&A]
Hi!
I have a predefined order for arranging the training examples in an `i`th epoch of my training loop (instead of random shuffling)

Apart from writing my own training loop, is there a simpler way to specify this order in the DataReader itself?

(Each epoch has got a different predefined order)

Thanks! :)

**Comment by epwalsh** [comment]
Hi @awasthiabhijeet, the best way to do this would be to define a custom [`BatchSampler`](https://github.com/allenai/allennlp/blob/27da04cf4252abb08c78a71eaacd76d0b766f66b/allennlp/data/samplers/batch_sampler.py#L7) for the [`DataLoader`](https://github.com/allenai/allennlp/blob/27da04cf4252abb08c78a71eaacd76d0b766f66b/allennlp/data/data_loaders/multiprocess_data_loader.py#L206).
